python/ray/dag/dag_node_operation.py

```diff
@@ -215,6 +227,25 @@ def is_nccl_write(self) -> bool:
         """
         return self.operation.type == _DAGNodeOperationType.WRITE and self.requires_nccl
 
+    @property
+    def is_nccl_read(self) -> bool:
+        """
+        Check if this is a NCCL read operation.
+        """
+        return self.operation.type == _DAGNodeOperationType.READ and self.requires_nccl
+
+    @property
+    def is_nccl_write_to_upstream_actor(self) -> bool:
+        """
+        Check if this is a NCCL write operation that writes to an actor
+        that is also one of its immediate upstream NCCL actors.
```

Comment:
I found this logic hard to understand. Reading the issue, would it work if we instead modified this to `is_nccl_write_to_peer(self, upstream_actor)`, and make sure the caller passes in the right actor?

Reply:
(There may be more than one upstream_actors, so we need to pass them all in.)
`is_nccl_write_to_upstream_actor` is a short hand and makes the caller code more concise. We can change it to the following at caller side (dag_node_operation.py:792) if you find it easier to understand:

Original:
`overlapped_schedule[j].is_nccl_write_to_upstream_actor`
New:
`overlapped_schedule[j].is_nccl_write_to_peer(overlapped_schedule[j].upstream_actors)`